### PR TITLE
localed: return error when setting a non-installed keymap

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by bluca** [comment]
> Get rid of nftw(). We should refuse to use such useless APIs on principle.

On this topic, I had a working alternative (intended for the sysext/extension_images work) that is temporarily shelved while I try to get the solution working with overlayfs first, but that I might push again, you can see a preview here:

https://github.com/bluca/systemd/commit/34fd90bc31f4c0b463ce23978712d61c83de8684

(following commit on the same branch is an example of a callback implementation)

File: src/shared/kbd-util.c [hunk 1]
```diff
@@ -31,52 +31,48 @@ static int nftw_cb(
...
-        }
+        const char *dir;
+        NULSTR_FOREACH(dir, KBD_KEYMAP_DIRS)
+                if (nftw(dir, nftw_cb, 20, FTW_PHYS) < 0) {
+                        if (errno != ENOENT)
+                                log_debug_errno(errno, "Failed to read keymap list from %s, ignoring: %m", dir);
+                }
 
-        l = set_get_strv(keymaps);
+        _cleanup_strv_free_ char **l = set_get_strv(keymaps);
```
Comment (bluca) [inline src/shared/kbd-util.c:77]: I've noticed in other PRs this trend to move declaration from the top of a function - nothing against it, but if we have a preference for this pattern, should we document it in the coding style?

File: src/shared/kbd-util.c [hunk 2]
```diff
@@ -31,52 +31,48 @@ static int nftw_cb(
...
+                return -1;
+        }
 
         e = endswith(p, ".map");
-        if (e)
-                *e = 0;
-
-        e = endswith(p, ".map.gz");
+        if (!e)
+                e = endswith(p, ".map.gz");
```
Comment (bluca) [inline src/shared/kbd-util.c]: nit: this change is unrelated to this commit, and is reworked anyway in the last commit, wouldn't it be cleaner to avoid it?

**Review by bluca — APPROVED** [review]
lgtm, just one minor nit

File: src/shared/kbd-util.c (hunk 1 above)
Comment (keszybz) [reply src/shared/kbd-util.c:77]: I don't think there's clear consensus. I prefer to have the variables tightly scoped, but Lennart likes to have everything defined at the top of the function…

File: src/shared/kbd-util.c (hunk 2 above)
Comment (keszybz) [reply src/shared/kbd-util.c]: The optimization in the last commit didn't initially occur to me. If it's OK, I would rather not rework this.

Comment (bluca) [reply src/shared/kbd-util.c]: sure no problem, it's just a minor thing and rebases are a pain
